core/arch/instrlist.c

```diff
@@ -245,7 +245,8 @@ check_translation(instrlist_t *ilist, instr_t *inst)
 #if defined(CLIENT_INTERFACE) && defined(ARM)
     if (instr_is_meta(inst)) {
         dr_pred_type_t auto_pred = ilist->auto_pred;
-        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_NONE) {
+        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_OP &&
```

Comment:
`DR_PRED_OP` is not a real predicate: it means that the predicate determines the opcode.  Yes it should be here for completeness but please use instr_predicate_is_cond() instead.  (Maybe we should export that and use it in drmemtrace too?)